### PR TITLE
feat(Day 8): Class-based components

### PR DESCRIPTION
**Pull request by ruchamahabal**
https://endearing-pony-0aaded.netlify.app/

- About Us page Demonstrating class-based component lifecycle with Parent (AboutUs), Child (AboutUsChildSection), and NestedChild components
   - constructor
   - render
   - componentDidMount
   - componentDidUpdate
   - componentWillUnmount
- Using and updating states
- Passing props

image

## componentWillUnmount demo using toggle

Kapture 2022-12-19 at 15 35 47

## Component lifecycle

Screen Shot 2022-12-19 at 3 29 34 PM

### CODE REVIEW

**Comment by ruchamahabal** [comment]
Deployed on netlify, not on github pages: https://endearing-pony-0aaded.netlify.app/

**Comment by ruchamahabal** [comment]
Note for the reviewer: Just check the `class_components` directory, readme and App.js. Other things are almost the same as before.

File: 8_class_components/src/App.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+// Since App.js is added as a module, we can import packages here using import statement
+import ReactDOM from "react-dom/client";
+import { createBrowserRouter, Outlet, RouterProvider } from "react-router-dom";
+
+import Navbar from "./components/Navbar.js";
+import MemberList from "./components/MemberList.js";
+import MemberDetails from "./components/MemberDetails";
+import Error from "./components/Error.js";
+
```
Comment (ap221882) [inline 8_class_components/src/App.js:11]: If trying to group your imports, can add one line comment that by which criteria you're putting them separately, i.e., Local, Global, UI, styles, etc. These things will help when you've dozens of imported modules.

File: 8_class_components/src/App.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+			},
+			{
+				path: "/about-us",
+				element: <AboutUs team_name="Web Pirates" />,
+				children: [
+					{
+						index: true,
+						element: (
+							<AboutUsChildSection
+								team_members="10"
```
Comment (ap221882) [inline 8_class_components/src/App.js:47]: Format the code here

File: 8_class_components/src/components/Card.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+				{location && (
+					<div className="detail-row">
+						<img className="icon" src={location_logo}></img>
+						<p className="location">{`${location}`}</p>
+					</div>
+				)}
+			</div>
+			<div className="card-footer social-icon">
+				<a href={html_url}>
+					<svg
```
Comment (ap221882) [inline 8_class_components/src/components/Card.js]: put svgs in separate file in assets folder. Keeps your code more clean.

File: 8_class_components/src/components/Error.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { useRouteError } from "react-router-dom";
+
+const Error = () => {
+	const errorData = useRouteError();
+	return (
+		<div className="centered-div">
+			<h2>Oops! Something went wrong!</h2>
+			<p>{`${errorData.statusText} : ${errorData.message}`}</p>
```
Comment (ap221882) [inline 8_class_components/src/components/Error.js]: Put an || operator, either of the value can come during the error

File: 8_class_components/src/components/Error.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { useRouteError } from "react-router-dom";
+
+const Error = () => {
+	const errorData = useRouteError();
```
Comment (ap221882) [inline 8_class_components/src/components/Error.js:4]: Have a blank line before return if your component involves some logic

File: 8_class_components/src/components/MemberDetails.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+const Member = () => {
+	let { username } = useParams();
+	const [member, setMember] = useState("");
+
+	if (!username) return;
+
+	useEffect(() => {
+		getMemberDetails();
+	}, []);
```
Comment (ap221882) [inline 8_class_components/src/components/MemberDetails.js:15]: add respective dependencies that are involved.

File: 8_class_components/src/components/SearchBar.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
+import { useState } from "react";
+
+/**
+ * returns filtered members that match name/designation with searchText
+ * @param {string} searchText text used to match for filtering
+ * @param {object} data data to filter from
+ */
```
Comment (ap221882) [inline 8_class_components/src/components/SearchBar.js:7]: can add @return too

File: 8_class_components/src/style.css [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+	justify-content: space-evenly;
+}
+
+/* About us */
+
+.emoji {
+	font-size: 50px;
+}
+
+button {
```
Comment (ap221882) [inline 8_class_components/src/style.css]: Add this html tag selector to the top having global style and then add custom styles using classes according to different scenarios of the respective button.

File: 8_class_components/src/components/class_components/NestedChild.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+		if (show_emoji) {
+			emoji = <Emoji />;
+		}
+
+		return (
+			<>
+				{course_name} course takes you from zero to hero in {duration}{" "}
+				months with hands-on coding and theory assignments,
+				<br /> interactive lectures, code reviews and a lot more!
```
Comment (ap221882) [inline 8_class_components/src/components/class_components/NestedChild.js]: can style through css, no need of br tag

**Review by ap221882 — COMMENTED** [review]
Very beautifully written. Commenting is awesome. Kudos Rucha!!

Edit1: CSS is also good

File: 8_class_components/src/App.js (hunk 2 above)
Comment (ruchamahabal) [reply 8_class_components/src/App.js:47]: Prettier auto-formatted it 🙈. Any suggestion?

File: 8_class_components/src/components/MemberDetails.js (hunk 6 above)
Comment (ruchamahabal) [reply 8_class_components/src/components/MemberDetails.js:15]: No dependencies to be specified in this case, I guess. Correct me if I am wrong.

**Comment by ruchamahabal** [comment]
thank you for your valuable feedback @ap221882 🙏🏻

**Comment by Pujarini** [comment]
Great looking UI ❤️ !

File: 8_class_components/src/components/Navbar.js [hunk 10]
```diff
@@ -1,11 +1,23 @@
...
 		<div className="navbar">
 			<div className="logo">
 				<img src={logo} alt="logo"></img>
-				<h1 className="title">Web Pirates</h1>
+				<h1 className="title link">Web Pirates</h1>
+			</div>
+
+			<div className="nav-links">
+				<Link to="/" className="link">
+					Home
```
Comment (Pujarini) [inline 8_class_components/src/components/Navbar.js:15]: can you use prettier for formatting?

Comment (ruchamahabal) [reply 8_class_components/src/components/Navbar.js:15]: Yes, prettier auto-formatted it this way

**Comment by ruchamahabal** [comment]
> Great looking UI ❤️ !

Thanks for reviewing ❤️

File: 8_class_components/src/App.js (hunk 2 above)
Comment (ap221882) [reply 8_class_components/src/App.js:47]: No comments 🥹

**Pull request merged**